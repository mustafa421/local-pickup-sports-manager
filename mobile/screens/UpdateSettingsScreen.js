import React, { Component } from "react";
import { StyleSheet, View, Platform, TextInput } from "react-native";
import { Button } from "react-native-elements";
import { connect } from "react-redux";
import PropTypes from "prop-types";
// import {Text, Platform } from "react-native";

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "stretch",
    justifyContent: "flex-start"
  }
});
class UpdateSettingsScreen extends Component {
  static navigationOptions = ({ navigation }) => ({
    title: "Edit",
    headerLeft: (
      <Button
        title="Cancel"
        textStyle={{ color: "rgba(0, 122, 255, 1)" }}
        onPress={() => navigation.navigate("settings")}
        backgroundColor="rgba(0,0,0,0)"
      />
    ),
    headerStyle: {
      marginTop: Platform.OS === "android" ? 24 : 0 // To prevent overlapping from header in Android devices
    }
  });

  constructor(props) {
    super(props);
    this.state = {};
  }

  // eslint-disable-next-line class-methods-use-this
  async onPressButton(state) {
    const { navigation, userID, email, username } = this.props;
    const obj = {
      userID,
      name: state.nameVal,
      email: state.emailVal
    };
    // console.log(obj);
    if (obj.email === undefined || obj.email === "") {
      obj.email = email;
    }
    if (obj.name === undefined || obj.name === "") {
      obj.name = username;
    }
    try {
      const request = await fetch(
        "http://local-pickup-sports-manager.herokuapp.com/editUser",
        {
          method: "POST",
          headers: {
            "Content-Type": "application/json"
          },
          body: JSON.stringify(obj),
          json: true
        }
      );
      console.log(obj);
      console.log(`[DEBUG] Server responded with ${request.status}`);
      console.log(await request.json());
      navigation.navigate("settings");
    } catch (err) {
      console.log(`Error sending request to server ${err}`);
      // Add alert to tell user that something went wrong
    }
  }

  render() {
    const { emailVal, nameVal } = this.state;
    return (
      <View style={styles.container}>
        <TextInput
          style={{ padding: 50, color: "grey" }}
          placeholder="Name"
          selectedValue={(this.state && nameVal) || "a"}
          onChangeText={text => this.setState({ nameVal: text })}
        />
        <TextInput
          style={{ padding: 50, color: "grey" }}
          placeholder="Email address"
          selectedValue={(this.state && emailVal) || "a"}
          onChangeText={text => this.setState({ emailVal: text })}
        />
        <Button
          title="Update"
          backgroundColor="blue"
          onPress={() => this.onPressButton(this.state)}
        />
      </View>
    );
  }
}
const mapStateToProps = state => state.auth.userAccountData;

export default connect(mapStateToProps)(UpdateSettingsScreen);

UpdateSettingsScreen.propTypes = {
  // eslint-disable-next-line react/no-unused-prop-types
  userID: PropTypes.number
};

UpdateSettingsScreen.defaultProps = {
  userID: null
};
