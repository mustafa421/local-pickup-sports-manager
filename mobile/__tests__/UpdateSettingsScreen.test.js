import React from "react";
import renderer from "react-test-renderer";
import UpdateSettingsScreen from "../screens/UpdateSettingsScreen";

describe("update settings screen tests", () => {
  test("snapshot", () => {
    const tree = renderer.create(<UpdateSettingsScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
