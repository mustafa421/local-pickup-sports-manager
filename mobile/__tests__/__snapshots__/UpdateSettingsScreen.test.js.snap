// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`update settings screen tests snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    defaultValue=""
    onChangeText={[Function]}
    placeholder="Name"
    style={
      Object {
        "color": "grey",
        "padding": 50,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    defaultValue=""
    onChangeText={[Function]}
    placeholder="Email address"
    style={
      Object {
        "color": "grey",
        "padding": 50,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    style={
      Array [
        Object {
          "marginLeft": 15,
          "marginRight": 15,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        pointerEvents="box-only"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "blue",
            },
            undefined,
            Object {
              "padding": 12,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Update
        </Text>
      </View>
    </View>
  </View>
</View>
`;
